lib/puppet/type/consul_prepared_query.rb

```diff
@@ -108,4 +108,18 @@
       raise ArgumentError, "Number of API tries must be a number" if not value.is_a?(Integer)
     end
   end
+
+  newparam(:template_regexp) do
+    desc 'regexp for template'
+    validate do |value|
```

Comment:
Simply defaulting to "nil" or whatever and then not injecting template data will make this "automatically" backwards compatible.